### ISSUE TITLE
report-unused-disable-directives reports false positive with eslint-disable-next-line no-fallthrough

### ISSUE THREAD

**Issue by backbone87** [eslint/eslint#13260]
**Tell us about your environment**

Node version: v12.5.0
npm version: v6.14.4
Local ESLint version: v6.8.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  root: true,
  env: {
    node: true,
  },
  extends: [
    'eslint:recommended',
    'plugin:prettier/recommended',
    'plugin:@typescript-eslint/eslint-recommended',
    'plugin:@typescript-eslint/recommended',
    'prettier/@typescript-eslint',
  ],
  rules: {
    'no-console': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',

    complexity: ['error', 12],
    'no-template-curly-in-string': 'warn',
    'no-return-await': 'error',
    '@typescript-eslint/no-use-before-define': ['error', { functions: false }],
    '@typescript-eslint/no-var-requires': 'off', // reenabled only for .tsx? files
    'padding-line-between-statements': [
      'error',
      // always blank line before control jumps
      { blankLine: 'always', prev: '*', next: ['break', 'continue', 'return', 'throw'] },
      // always blank line after directive & import blocks
      { blankLine: 'always', prev: ['directive', 'import'], next: '*' },
      { blankLine: 'any', prev: 'directive', next: 'directive' },
      { blankLine: 'any', prev: 'import', next: 'import' },
    ],
  },
  overrides: [
    {
      files: ['*.ts'],
      rules: {
        '@typescript-eslint/explicit-function-return-type': [
          'error',
          { allowExpressions: true, allowTypedFunctionExpressions: true },
        ],
        '@typescript-eslint/no-explicit-any': 'off',
        '@typescript-eslint/no-parameter-properties': [
          'error',
          {
            // we actually allow all parameter props except "readonly" without
            // visibility modifier, this is inline with normal prop declarations
            allows: ['private', 'protected', 'public', 'private readonly', 'protected readonly', 'public readonly'],
          },
        ],
        '@typescript-eslint/no-require-imports': 'error',
        '@typescript-eslint/no-var-requires': 'error',
        '@typescript-eslint/explicit-member-accessibility': 'error',
      },
    },
  ],
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
      // in order to prevent the browser from opening a basic auth dialog we allow
      // "Credentials" to be used as an auth scheme and do not send the
      // "WWW-Authenticate" header, if this scheme is used
      case 'Credentials':
        authorization = `Basic ${authorization.slice(12)}`;
        realm = undefined;

      // eslint-disable-next-line no-fallthrough
      case 'Basic': {
        const credentials = parse(authorization);
        if (!credentials) {
          throw new AuthenticationRequired(realm);
        }

        let tokenSet;
        try {
          tokenSet = await this.oidcClient.grant({
            ['client_id']: this.oidcClient.metadata.client_id,
            ['grant_type']: 'password',
            username: credentials.name,
            password: credentials.pass,
            scope: 'openid',
          });
        } catch {
          throw new AuthenticationRequired(realm);
        }

        const token = tokenSet.id_token;
        if (!token) {
          throw new Error('ID token missing! Should never happen with grant type "password"!');
        }

        return { token, claims: tokenSet.claims() };
      }
```

```bash
npx eslint --report-unused-disable-directives "./{src,tests}/**/*.{js,ts}"
```

**What did you expect to happen?**
Everything ok, no unused directives

**What actually happened? Please include the actual, raw output from ESLint.**
```
PATH/TO/authenticator.ts
  61:7  error  Unused eslint-disable directive (no problems were reported from 'no-fallthrough')
```

removing the `// eslint-disable-next-line no-fallthrough` results in:
```
PATH/TO/authenticator.ts
  61:7  error  Expected a 'break' statement before 'case'  no-fallthrough

✖ 1 problem (1 error, 0 warnings)
```
using `// eslint-disable no-fallthrough` & `// eslint-enable no-fallthrough` around the entire case block works as expected

**Are you willing to submit a pull request to fix this bug?**
idk, what the failure source is

**Comment by anikethsaha** [comment]
I think this is because ` no-fallthrough` was never reported for `case 'basic':` because it has a return statement at the end and this rule does not reports for that.  

[refer this](https://eslint.org/docs/rules/no-fallthrough#when-not-to-use-it)

**Comment by backbone87** [comment]
but when i remove the `// eslint-disable-next-line no-fallthrough` the rule does report an error (cli and via vscode), so the rule clearly triggers on that case

**Comment by anikethsaha** [comment]
this is because of the case `Credentials`. the error is in `Credentials` so you need to use `no-fallthrough` in that case instead of in `basic`. the `Basic` case is working as expected and is lint-free so when you are using disable comment, it reports because it was never reported. 

[demo](https://eslint.org/demo#eyJ0ZXh0IjoiZnVuY3Rpb24gZm9vICgpe1xuXG5cbnN3aXRjaChkKXtcblxuY2FzZSAnQ3JlZGVudGlhbHMnOlxuICAgICAgICBhdXRob3JpemF0aW9uID0gYEJhc2ljICR7YXV0aG9yaXphdGlvbi5zbGljZSgxMil9YDtcbiAgICAgICAgcmVhbG0gPSB1bmRlZmluZWQ7XG5cdFx0Ly8gbm8tZmFsbHRocm91Z2hcbiAgICBcbiAgICAgIGNhc2UgJ0Jhc2ljJzoge1xuICAgICAgICBjb25zdCBjcmVkZW50aWFscyA9IHBhcnNlKGF1dGhvcml6YXRpb24pO1xuICAgICAgICBpZiAoIWNyZWRlbnRpYWxzKSB7XG4gICAgICAgICAgdGhyb3cgbmV3IEF1dGhlbnRpY2F0aW9uUmVxdWlyZWQocmVhbG0pO1xuICAgICAgICB9XG5cbiAgICAgICAgbGV0IHRva2VuU2V0O1xuICAgICAgICB0cnkge1xuICAgICAgICAgIHRva2VuU2V0ID0gdGhpcy5vaWRjQ2xpZW50LmdyYW50KHtcbiAgICAgICAgICAgIFsnY2xpZW50X2lkJ106IHRoaXMub2lkY0NsaWVudC5tZXRhZGF0YS5jbGllbnRfaWQsXG4gICAgICAgICAgICBbJ2dyYW50X3R5cGUnXTogJ3Bhc3N3b3JkJyxcbiAgICAgICAgICAgIHVzZXJuYW1lOiBjcmVkZW50aWFscy5uYW1lLFxuICAgICAgICAgICAgcGFzc3dvcmQ6IGNyZWRlbnRpYWxzLnBhc3MsXG4gICAgICAgICAgICBzY29wZTogJ29wZW5pZCcsXG4gICAgICAgICAgfSk7XG4gICAgICAgIH0gY2F0Y2gge1xuICAgICAgICAgIHRocm93IG5ldyBBdXRoZW50aWNhdGlvblJlcXVpcmVkKHJlYWxtKTtcbiAgICAgICAgfVxuXG4gICAgICAgIGNvbnN0IHRva2VuID0gdG9rZW5TZXQuaWRfdG9rZW47XG4gICAgICAgIGlmICghdG9rZW4pIHtcbiAgICAgICAgICB0aHJvdyBuZXcgRXJyb3IoJ0lEIHRva2VuIG1pc3NpbmchIFNob3VsZCBuZXZlciBoYXBwZW4gd2l0aCBncmFudCB0eXBlIFwicGFzc3dvcmRcIiEnKTtcbiAgICAgICAgfVxuXG4gICAgICAgIHJldHVybiB7IHRva2VuLCBjbGFpbXM6IHRva2VuU2V0LmNsYWltcygpIH07XG4gICAgICB9XG59IFxuXG5cbn1cbiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTEsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsibm8tZXh0cmEtcGFyZW5zIjoyLCJuby1mYWxsdGhyb3VnaCI6Mn0sImVudiI6e319fQ==)

**Comment by yeonjuan** [comment]
Ht @backbone87  :) thanks for the reporting 😀 

I think there is a conflict between **no-fallthrough**'s default [commentPattern](https://eslint.org/docs/rules/no-fallthrough#commentpattern) and the eslint directives.

The **no-fallthrough** use default commentPattern -`/falls?\s?through/i`.
And it seems **no-fallthrough**  treats the directive(`// eslint-disable-next-line no-fallthrough`) as a comment for intentional using (because the directive contains the string "fallthrough" so it caught by the default commentPattern)

So, It seems a bug to me. but  I want to hear about other member's opinions.

**Comment by anikethsaha** [comment]
ummm....yes, that can be conflicting, I suppose we can change the default pattern to this 

`^[\s]*falls?\s?through`

**Comment by backbone87** [comment]
@yeonjuan that makes sense

however i wonder why the rule does provide its own way to "skip" it and does not rely on `// eslint-disable` comments. i know some probably think using disable directives is evil, but imho its the exact same effect. usually i expect a comment why rule was skipped and its easier to validate that all disable directives have proper comments
